### DISCUSSION TITLE
Correct way to measure CPU & GPU usage in PIXI app.

### DISCUSSION THREAD

**Discussion by humbleCoder1990** [pixijs/pixijs#7738] [Q&A]
![image](https://user-images.githubusercontent.com/87828192/130366018-f194ffa2-1a62-421e-ab64-8f91a4c28266.png)

I use activity monitor on Mac to check cpu & gpu usages to check my PIXI app.

One thing I'm confusing is,  

Google Chrome Helper (GPU) is also using CPU. Is it gpu actaully?

If my  Google Chrome Helper (Renderer) is using 50% cpu,
Then, Google Chrome Helper (GPU) is using 50% cpu, too
Is it 100% cpu usage?

How to measure cpu & gpu usage correctly?

**Comment by eXponenta** [comment]
For Debugging you should use a performance panel or profiler.

There are not a way measuring a GPU/CPU inside JS fully correctly for in-app profiling.
CPU + GPU API can be estimated by frame time approximation.

GPU run (how long a GPU really works) cannot be measured for WebGL (there was a timers API, but it was removed because some GPU is bugged).

WebGPU (pixi not support it) has some api for measuring a GPU working time.


Using a activity monitor is bad idea because it show full usage of devices.
This include a browser working threads, paints operations, CSS calculation etc and can be used relative a some zero-point state of application.

**Comment by ivanpopelyshev** [comment]
Which macbook is it?

**Reply by humbleCoder1990** [reply]
I'm using 13 Inch M1 macbook pro. (8GB)

**Reply by ivanpopelyshev** [reply]
I dont know how browser works with gpu on this architecture, so there might be that high CPU % is needed for any webgl app
